src/test/java/com/amazonaws/services/kinesis/multilang/MultiLangDaemonConfigTest.java

```diff
@@ -88,4 +102,126 @@ public void propertyValidation() {
+
+        try {
+            config = new MultiLangDaemonConfig(FILENAME, classLoader, buildMockConfigurator());
+            assertAgainstKclConfig(config.getKinesisClientLibConfiguration(), "http://proxy.com", 1234);
+        } catch (IllegalArgumentException e) {
+            // Bad
+        } catch (IOException e) {
+            Assert.fail();
+        }
+    }
+
+    @Test public void KinesisClientLibConfiguration_should_get_proxy_info_from_env_vars() {
+        String PROPERTIES =
+                "executableName = randomEXE \n" + "applicationName = testApp \n" + "streamName = fakeStream \n"
+                        + "AWSCredentialsProvider = DefaultAWSCredentialsProviderChain\n"
+                        + "processingLanguage = malbolge";
+        ClassLoader classLoader = Mockito.mock(ClassLoader.class);
+
+        Map<String, String> env = new HashMap<>();
+        env.put("http_proxy", "http://proxy.com:1234");
+
+        try {
+            setEnv(env);
+
+            Mockito.doReturn(new ByteArrayInputStream(PROPERTIES.getBytes())).when(classLoader)
+                    .getResourceAsStream(FILENAME);
+
+            MultiLangDaemonConfig config;
+
+            try {
+                config = new MultiLangDaemonConfig(FILENAME, classLoader, buildMockConfigurator());
+
+                // host doesn't have protocol here because URI splits that into a separate piece
+                assertAgainstKclConfig(config.getKinesisClientLibConfiguration(), "proxy.com", 1234);
+            } catch (IllegalArgumentException e) {
+                // Bad
+            } catch (IOException e) {
+                Assert.fail();
+            }
+        } catch (Exception e) {
+            Assert.fail();
+        }
+    }
+
+    @Test public void KinesisClientLibConfiguration_should_not_get_proxy_info() {
+        String PROPERTIES =
+                "executableName = randomEXE \n" + "applicationName = testApp \n" + "streamName = fakeStream \n"
+                        + "AWSCredentialsProvider = DefaultAWSCredentialsProviderChain\n"
+                        + "processingLanguage = malbolge";
+        ClassLoader classLoader = Mockito.mock(ClassLoader.class);
+
+        Map<String, String> env = new HashMap<>();
+
+        try {
+            //clear out any env vars loaded from system
+            setEnv(env);
+
+            Mockito.doReturn(new ByteArrayInputStream(PROPERTIES.getBytes())).when(classLoader)
+                    .getResourceAsStream(FILENAME);
+
+            MultiLangDaemonConfig config;
+
+            try {
+                config = new MultiLangDaemonConfig(FILENAME, classLoader, buildMockConfigurator());
+
+                // host doesn't have protocol here because URI splits that into a separate piece
+                assertAgainstKclConfig(config.getKinesisClientLibConfiguration(), null, -1);
+            } catch (IllegalArgumentException e) {
+                // Bad
+            } catch (IOException e) {
```

Comment:
No need to catch exceptions if you are going to call fail. You can let the exception be thrown from the test. The test will fail if exception is thrown.

Reply:
I have to catch the exception from the setEnv method, which is why that is there.

Reply:
Junit will fail the test if the test throws an exception, and if it is not expected.

```java
@Test
public void testScenario() throws Exception {
    // test goes here
}
```

If the above test throws an exception, Junit will fail the test.